packages/components/src/ecosystem_bar/index.jsx

```diff
@@ -0,0 +1,43 @@
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { useRef } from 'react';
+import { useOnceSwipeTracking, useScrolled } from '@/src/ecosystem_bar/hooks';
+import styles from './index.module.scss';
+
+const EcosystemBar = ({ items, onFirstSwipe, onItemClick }) => {
+  const navRef = useRef(null);
+  const scrolled = useScrolled(navRef);
+
+  useOnceSwipeTracking(navRef, onFirstSwipe);
+
+  return (
+    <ul className={clsx(styles.root, { [styles.scrolled]: scrolled })} ref={navRef}>
+      {items.map((item) => (
```

Comment:
We usually put map function into a separate render function, though it won't really improve readability much, but will be consistent though.